### PR TITLE
chore: upgrade react-router-dom to v6

### PR DESCRIPTION
**Pull request by chenxsan**
This pull request would upgrade `react-router-dom` to its latest v6.

- [x] Migrate `<Redirect />` to `<Navigate replace />`, see https://reactrouter.com/docs/en/v6/upgrading/v5#use-usenavigate-instead-of-usehistory
- [x] Migrate `<Switch />` to `<Routes />`, see https://reactrouter.com/docs/en/v6/upgrading/v5#upgrade-all-switch-elements-to-routes
- [x] Use `useNavigate` instead of `useHistory`, see https://reactrouter.com/docs/en/v6/upgrading/v5#use-usenavigate-instead-of-usehistory
- [x] Use `<Route element={} />` instead of `<Route children={} />`

There are some upstream issues, e.g.:

1. https://github.com/remix-run/react-router/issues/8381 We can't use optional parameters in v6, e.g., `path='/add/:tokenA/:tokenB?'` is not supported
2. https://github.com/remix-run/react-router/issues/8161 No more `activeClassName`, using function for `className` isn't supported by styled-components

But I did find workaround for them.

### CODE REVIEW

**Comment by chenxsan** [comment]
Regarding the failed `Test / deps-tests`:

```
yarn why @babel/runtime
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "@babel/runtime"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "@babel/runtime@7.17.0"
info Has been hoisted to "@babel/runtime"
info Reasons this module exists
   - Hoisted from "react-window#@babel#runtime"
   - Hoisted from "aria-query#@babel#runtime"
   - Hoisted from "@lingui#core#@babel#runtime"
   - Hoisted from "@lingui#macro#@babel#runtime"
   - Hoisted from "@lingui#react#@babel#runtime"
   - Hoisted from "@lingui#cli#@babel#runtime"
   - Hoisted from "react-redux#@babel#runtime"
   - Hoisted from "polished#@babel#runtime"
   - Hoisted from "@testing-library#react#@babel#runtime"
   - Hoisted from "react-spring#@babel#runtime"
   - Hoisted from "babel-plugin-macros#@babel#runtime"
   - Hoisted from "redux#@babel#runtime"
   - Hoisted from "@testing-library#jest-dom#@babel#runtime"
   - Hoisted from "@reach#dialog#react-focus-lock#@babel#runtime"
   - Hoisted from "@lingui#cli#@lingui#babel-plugin-extract-messages#@babel#runtime"
   - Hoisted from "@lingui#macro#@lingui#conf#@babel#runtime"
   - Hoisted from "react-scripts#eslint-plugin-jsx-a11y#@babel#runtime"
   - Hoisted from "@types#testing-library__cypress#@testing-library#dom#@babel#runtime"
   - Hoisted from "@testing-library#react#@testing-library#dom#@babel#runtime"
   - Hoisted from "@lingui#cli#babel-plugin-macros#@babel#runtime"
   - Hoisted from "@coinbase#wallet-sdk#eth-block-tracker#@babel#runtime"
   - Hoisted from "react-scripts#workbox-webpack-plugin#@babel#runtime"
   - Hoisted from "@reach#dialog#react-focus-lock#react-clientside-effect#@babel#runtime"
   - Hoisted from "rebass#reflexbox#@emotion#core#@babel#runtime"
   - Hoisted from "node-vibrant#@vibrant#image-node#@jimp#custom#@babel#runtime"
   - Hoisted from "node-vibrant#@vibrant#image-node#@jimp#plugin-resize#@babel#runtime"
   - Hoisted from "node-vibrant#@vibrant#image-node#@jimp#types#@babel#runtime"
   - Hoisted from "react-scripts#workbox-webpack-plugin#workbox-build#@babel#runtime"
   - Hoisted from "@graphql-codegen#typescript#@graphql-codegen#visitor-plugin-common#@graphql-tools#relay-operation-optimizer#relay-compiler#@babel#runtime"
   - Hoisted from "rebass#reflexbox#@emotion#styled#@emotion#styled-base#@babel#runtime"
   - Hoisted from "node-vibrant#@vibrant#image-node#@jimp#types#@jimp#bmp#@babel#runtime"
   - Hoisted from "node-vibrant#@vibrant#image-node#@jimp#custom#@jimp#core#@babel#runtime"
   - Hoisted from "node-vibrant#@vibrant#image-node#@jimp#types#@jimp#gif#@babel#runtime"
   - Hoisted from "node-vibrant#@vibrant#image-node#@jimp#types#@jimp#jpeg#@babel#runtime"
   - Hoisted from "node-vibrant#@vibrant#image-node#@jimp#types#@jimp#png#@babel#runtime"
   - Hoisted from "node-vibrant#@vibrant#image-node#@jimp#types#@jimp#tiff#@babel#runtime"
   - Hoisted from "node-vibrant#@vibrant#image-node#@jimp#plugin-resize#@jimp#utils#@babel#runtime"
   - Hoisted from "@graphql-codegen#typescript#@graphql-codegen#visitor-plugin-common#@graphql-tools#relay-operation-optimizer#relay-compiler#relay-runtime#@babel#runtime"
   - Hoisted from "react-scripts#babel-preset-react-app#@babel#preset-env#@babel#plugin-transform-regenerator#regenerator-transform#@babel#runtime"
info Disk size without dependencies: "844KB"
info Disk size with unique dependencies: "888KB"
info Disk size with transitive dependencies: "888KB"
info Number of shared dependencies: 1
=> Found "babel-preset-react-app#@babel/runtime@7.12.1"
info This module exists because "react-scripts#babel-preset-react-app" depends on it.
info Disk size without dependencies: "708KB"
info Disk size with unique dependencies: "752KB"
info Disk size with transitive dependencies: "752KB"
info Number of shared dependencies: 1
=> Found "history#@babel/runtime@7.18.9"
info This module exists because "react-router-dom#history" depends on it.
info Disk size without dependencies: "880KB"
info Disk size with unique dependencies: "924KB"
info Disk size with transitive dependencies: "924KB"
info Number of shared dependencies: 1
✨  Done in 1.06s.
```

**Comment by vm** [comment]
I think I fixed the conflicts/test issues, so let me know when you'd like a review!

**Comment by chenxsan** [comment]
> I think I fixed the conflicts/test issues, so let me know when you'd like a review!

Awesome! Will let u know when it's ready.

**Comment by chenxsan** [comment]
@vm I think it's ready for reviewing. There's still one failing e2e test, but it doesn't seem related to my changes.

File: cypress/e2e/wallet.test.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { TEST_ADDRESS_NEVER_USE_SHORTENED } from '../support/ethereum'
 
 describe('Wallet', () => {
   before(() => {
-    cy.visit('/')
+    cy.visit('/swap')
```
Comment (chenxsan) [inline cypress/e2e/wallet.test.ts:5]: Tests become flaky when there's a redirection here, so I update it to visit `/swap` path directly.

Comment (vm) [reply cypress/e2e/wallet.test.ts:5]: ah yeah that works

**Review by vm — APPROVED** [review]
this LGTM, I will do some further testing later this week and merge. thank you so much for this work, we appreciate it!!

**Pull request merged**